File: source/sort.py
import numpy as np
import time

def process(data, outfile, test_index):
    start = time.time()
    sorted_data = np.sort(data)
    end = time.time()
    duration = (end - start) * 1000 
    outfile.write(f"Sorting time {test_index}: {duration:.3f} ms\n")

def main():
    with open("test.txt", "r") as infile, open("time.txt", "w") as outfile:
        test_index = 1
        for line in infile:
            data = np.array(list(map(float, line.split())))
            process(data, outfile, test_index)
            test_index += 1

if __name__ == "__main__":
    main()

File: source/sort.cpp
#include <bits/stdc++.h>
using namespace std;
using ld = long double;

void process(vector<ld> &data, ofstream &outfile, int testIndex) {
    auto start = chrono::high_resolution_clock::now();
    sort(data.begin(), data.end());
    auto end = chrono::high_resolution_clock::now();
    chrono::duration<double, milli> duration = end - start;
    outfile << "Sorting time " << testIndex << ": " << duration.count() << " ms\n";
}

int main() {
    ifstream infile("test.txt");
    ofstream outfile("time.txt");
    string line;
    int testIndex = 1;
    while (getline(infile, line)) {
        stringstream ss(line);
        vector<ld> data;
        ld num;
        while (ss >> num) {
            data.push_back(num);
        }
        process(data, outfile, testIndex++);
    }
    return 0;
}
